### PR TITLE
threads: linux: simplify setting thread priority

### LINKED ISSUE #22755: Nexus immediately crashes on startup
### Describe the bug
```
% kodi-build/kodi.bin -p --standalone
/usr/include/c++/12.2.1/bits/stl_algo.h:3623: constexpr const _Tp& std::clamp(const _Tp&, const _Tp&, const _Tp&) [with _Tp = int]: Assertion '!
(__hi < __lo)' failed.
[1]    246845 IOT instruction (core dumped)
```

```
(gdb) run -p --standalone
Starting program: /home/plv/tmp/kodi/trunk/src/kodi-build/kodi.bin -p --standalone

[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[Detaching after vfork from child process 247449]
[New Thread 0x7ffff15a66c0 (LWP 247451)]
/usr/include/c++/12.2.1/bits/stl_algo.h:3623: constexpr const _Tp& std::clamp(const _Tp&, const _Tp&, const _Tp&) [with _Tp = int]: Assertion '!
(__hi < __lo)' failed.
[New Thread 0x7ffff0da56c0 (LWP 247452)]

Thread 2 "Announce" received signal SIGABRT, Aborted.
[Switching to Thread 0x7ffff15a66c0 (LWP 247451)]
0x00007ffff62d98ec in ?? () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff62d98ec in  () at /usr/lib/libc.so.6
#1  0x00007ffff628aea8 in raise () at /usr/lib/libc.so.6
#2  0x00007ffff627453d in abort () at /usr/lib/libc.so.6
#3  0x00007ffff5ad30e2 in std::__glibcxx_assert_fail(char const*, int, char const*, char const*)
    (file=<optimized out>, line=<optimized out>, function=<optimized out>, condition=<optimized out>)
    at /usr/src/debug/gcc/gcc/libstdc++-v3/src/c++11/debug.cc:60
#4  0x000055555611bd30 in CThreadImplLinux::SetPriority(ThreadPriority const&) ()
#5  0x0000555556f37886 in ANNOUNCEMENT::CAnnouncementManager::Process() ()
#6  0x0000555556570005 in CThread::Action() ()
#7  0x000055555657171f in CThread::Create(bool)::{lambda(CThread*, std::promise<bool>)#1}::operator()(CThread*, std::promise<bool>) const [clone
 .constprop.0] ()
#8  0x00005555565722cf in std::thread::_State_impl<std::thread::_Invoker<std::tuple<CThread::Create(bool)::{lambda(CThread*, std::promise<bool>)
#1}, CThread*, std::promise<bool> > > >::_M_run() ()
#9  0x00007ffff5ad72c3 in std::execute_native_thread_routine(void*) (__p=0x555559400120)
    at /usr/src/debug/gcc/gcc/libstdc++-v3/src/c++11/thread.cc:82
#10 0x00007ffff62d7bb5 in  () at /usr/lib/libc.so.6
#11 0x00007ffff6359d90 in  () at /usr/lib/libc.so.6
(gdb)
```

### Debuglog
No log produced.

### Your Environment
Used Operating system:
 - [x] Linux

 - Operating system version/name: Arch Linux 6.1.11-arch1-1
 - Kodi version: 20

### PR DESCRIPTION
**Pull request by lrusak**
This is my attempt at fixing #22755

Instead of trying to fix the mess of inverted niceness and clamping let's just simplify the entire thing.

This just tries to set a priority level (based on the app priority level). If it works great, if not the app priority will be used (the default).

Some background:

On linux there is a niceness level which sets the priority. It's an inverted property from -20 (highest priority) to 20 (lowest priority). Setting a higher priority than `0` typically requires changing `limits.conf` (not obvious). A root user can set any priority.

Since we don't do anything (read "can't"), there is no point in trying to calculate the maximum allowable priority level. We can just blindly set the priority and then read back the priority of the thread. This simplifies the (very confusing) logic of trying to determine the max priority.

example:
```
$ ./kodi.bin --logging=console | grep 'priority:' 

2023-05-01 21:10:32.200 T:3802957   debug <general>: [threads] app priority: '0'
2023-05-01 21:10:32.200 T:3802957   debug <general>: [threads] name: 'Announce' priority: '1'
2023-05-01 21:10:32.757 T:3802955   debug <general>: [threads] name: 'libinput' priority: '1'
2023-05-01 21:10:33.269 T:3802992   debug <general>: [threads] name: 'AESink' priority: '0'
2023-05-01 21:10:33.300 T:3802994   debug <general>: [threads] name: 'JobWorker' priority: '1'
2023-05-01 21:10:33.524 T:3802995   debug <general>: [threads] name: 'JobWorker' priority: '1'
2023-05-01 21:10:33.524 T:3802996   debug <general>: [threads] name: 'JobWorker' priority: '1'
2023-05-01 21:10:33.604 T:3802955   debug <general>: [threads] name: 'PeripBusUSBUdev' priority: '1'
2023-05-01 21:10:33.604 T:3802955   debug <general>: [threads] name: 'PeripBusAddon' priority: '1'
2023-05-01 21:10:34.668 T:3803039   debug <general>: [threads] name: 'JobWorker' priority: '1'
```

The inverted priority means that, `1` is lower priority than the app `0`. Higher priority (if allowed) would be `-1`.

You can also see this with `top`:
```
$ top -H -b -n1 -p $(pidof kodi.bin) 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                                                                                                                                                                                                                                          
3803385 lukas     25   5 3654736 406532 144072 S   0.3   1.2   0:00.81 kodi.bin                                                                                                                                                              
3803517 lukas     25   5 3654736 406532 144072 R   0.3   1.2   0:00.06 AESink                                                                                                                                                                
3803530 lukas     25   5 3654736 406532 144072 S   0.3   1.2   0:00.03 PeripEventScan                                                                                                                                                        
3803535 lukas     25   5 3654736 406532 144072 S   0.3   1.2   0:00.27 LanguageInvoker                                                                                                                                                       
3803434 lukas     26   6 3654736 406532 144072 S   0.0   1.2   0:00.00 Announce                                                                                                                                                              
3803438 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.10 kodi.bin                                                                                                                                                              
3803439 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 pipewire                                                                                                                                                              
3803480 lukas     26   6 3654736 406532 144072 S   0.0   1.2   0:00.00 libinput                                                                                                                                                              
3803493 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 kodi.bin                                                                                                                                                              
3803494 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 kodi.bin                                                                                                                                                              
3803495 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 kodi.bin                                                                                                                                                              
3803496 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 kodi.bin                                                                                                                                                              
3803497 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 kodi.bin                                                                                                                                                              
3803498 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 kodi.bin                                                                                                                                                              
3803499 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 kodi.bin                                                                                                                                                              
3803500 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 kodi.bin                                                                                                                                                              
3803501 lukas     39  19 3654736 406532 144072 S   0.0   1.2   0:00.00 kodi.bi:disk$0                                                                                                                                                        
3803516 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 ActiveAE                                                                                                                                                              
3803519 lukas     26   6 3654736 406532 144072 S   0.0   1.2   0:00.24 JobWorker                                                                                                                                                             
3803520 lukas     26   6 3654736 406532 144072 S   0.0   1.2   0:00.21 JobWorker                                                                                                                                                             
3803521 lukas     26   6 3654736 406532 144072 S   0.0   1.2   0:00.19 JobWorker                                                                                                                                                             
3803528 lukas     26   6 3654736 406532 144072 S   0.0   1.2   0:00.41 PeripBusUSBUdev                                                                                                                                                       
3803529 lukas     26   6 3654736 406532 144072 S   0.0   1.2   0:00.00 PeripBusAddon                                                                                                                                                         
3803531 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 Timer                                                                                                                                                                 
3803532 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:01.05 LanguageInvoker                                                                                                                                                       
3803533 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.35 LanguageInvoker                                                                                                                                                       
3803534 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.42 LanguageInvoker                                                                                                                                                       
3803537 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 EventServer                                                                                                                                                           
3803538 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 TCPServer                                                                                                                                                             
3803541 lukas     26   6 3654736 406532 144072 S   0.0   1.2   0:00.48 JobWorker                                                                                                                                                             
3803547 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.01 LanguageInvoker                                                                                                                                                       
3803552 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 LanguageInvoker                                                                                                                                                       
3803553 lukas     25   5 3654736 406532 144072 S   0.0   1.2   0:00.00 LanguageInvoker  
```

Sorry for the large ugly commit. It's probably be to view in the side by side view -> [link](https://github.com/xbmc/xbmc/pull/23227/files?diff=split&w=0)

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->

Fixes crashes on some systems.

### CODE REVIEW

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

File: xbmc/platform/linux/threads/ThreadImplLinux.cpp [hunk 1]
```diff
@@ -129,43 +83,23 @@ void CThreadImplLinux::SetThreadInfo(const std::string& name)
...
-      CLog::Log(LOGERROR, "[threads] failed to set priority: {}", strerror(errno));
-  }
+  m_name = name;
 }
 
 bool CThreadImplLinux::SetPriority(const ThreadPriority& priority)
 {
-  // keep priority in bounds
+  std::call_once(flag,
+                 [this]() { CLog::Log(LOGDEBUG, "[threads] app priority: '{}'", s_appPriority); });
```
Comment (neo1973) [inline xbmc/platform/linux/threads/ThreadImplLinux.cpp:92]:
```suggestion
                 []() { CLog::Log(LOGDEBUG, "[threads] app priority: '{}'", s_appPriority); });
```
Clang complains about the unused capture:
```
.../xbmc/platform/linux/threads/ThreadImplLinux.cpp:92:19: warning: lambda capture 'this' is not used [-Wunused-lambda-capture]
                 [this]() { CLog::Log(LOGDEBUG, "[threads] app priority: '{}'", s_appPriority); });
                  ^~~~
```

**Review by neo1973 — CHANGES_REQUESTED** [review]
Other than the warning it LGTM!

File: xbmc/platform/linux/threads/ThreadImplLinux.cpp (hunk 1 above)
Comment (lrusak) [reply xbmc/platform/linux/threads/ThreadImplLinux.cpp]: Thanks!

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
squashed the fixup

**Review by neo1973 — APPROVED** [review]

**Pull request merged**